andela/ah-robotics-frontend
# #16294904 A user can rate an article

#### What does this PR do?

Enables a user to rate the article.

#### Description of Task to be completed?
- Create a UI component for ratings
- Consume the Rating API endpoint
- Write tests to the feature

#### How should this be manually tested?
1. Navigate to a folder where you would prefer to test the repository.

2. Create a directory
 ```$```  ```mkdir myproject```

3. Navigate to the directory "myproject"
 ```$```   ```cd myproject```
4. Clone the author's haven repository
  ```$```   ```git clone https://github.com/andela/ah-robotics-frontend.git```
5. Change the directory into the "ah-robotics-fronted"
  ```$```   ```cd ah-robotics-frontend```

6.  Checkout to the branch ft-rate-article-162949041
 ```$```   ```git checkout ft-rate-article-162949041```

7. Install dependancies from the package.json
 ```$```   ```npm install``` to install all the modules listed in the `package.json` file.

8. Start the application in development mode
  ```$```   ```npm start``` to run the app in the development mode 

The application opens http://localhost:3000 in the browser.
Change the URL on the browser to;
 ```
https://ah-robotics-frontend-sta-pr-17.herokuapp.com
```

The homepage will appear as shown below.
![image](https://user-images.githubusercontent.com/14871749/53410237-42ea1d80-39d4-11e9-9abb-d6b9b620b88d.png)


9. Click the ```Login``` link on the Navbar and enter the following details to log in:
 **(for demo purpose use)**

**Username:** ```o.ogeto@yahoo.com``` 
**Password:** ```Ogeto1234$```

10. Once you are logged in, click The title of one article. You will be presented with the articles page below. Click the stars to rate the article.
![image](https://user-images.githubusercontent.com/14871749/53412013-36b48f00-39d9-11e9-83ac-f55aa7b764fd.png)


11. The system will process the ratings and return the average rating.

13. To run tests, open the terminal and run the following command:
```
npm test
``` 
14. To run tests and coverage, open the terminal and run the following command:
```
npm test -- --coverage
```
#### Any background context you want to provide?
The rating module takes in a predefined slug of an article and rates that article.

#### What are the relevant pivotal tracker stories?
[#162949041](https://www.pivotaltracker.com/story/show/162949041)

#### Screenshots (if appropriate)
##### Login a user
<img width="1280" alt="screen shot 2019-02-22 at 09 46 27" src="https://user-images.githubusercontent.com/14871749/53224753-d515bd00-3686-11e9-9aec-94eddaea401e.png">

##### Articles page
![image](https://user-images.githubusercontent.com/14871749/53412013-36b48f00-39d9-11e9-83ac-f55aa7b764fd.png)

#### When a user submits ratings:
![image](https://user-images.githubusercontent.com/14871749/53412152-a0cd3400-39d9-11e9-8278-c757b853b00f.png)


#### Questions:
